File: RiderQc.Ng/src/app/model/ride.ts
import { Comment } from '../model/comment';
import { User } from '../model/user';
import { Level } from '../model/level';
import { Trajet } from '../model/trajet';

export class Ride {
  RideId: number;
  Title: string ;
  Description: string ;
  CreatorId: number ;
  TrajetId: number ;
  LevelId: number ;
  DateDepart: Date ;
  DateFin: Date; 
  Creator: User;
  Level: Level;
  Trajet: Trajet;
  Comments: Comment[];
  Participants: string[];
  RideRating: number;

  constructor(RideId?: number, Title?: string, Description?: string, CreatorId?: number, TrajetId?: number, LevelId?: number, DateDepart?: Date, DateFin?: Date) {
      this.RideId = RideId;
      this.Title = Title;
      this.Description = Description;
      this.CreatorId = CreatorId;
      this.TrajetId = TrajetId;
      this.LevelId = LevelId;
      this.DateDepart = DateDepart;
      this.DateFin = DateFin;
  }
}

File: RiderQc.Web/Entities/Ride.cs
namespace RiderQc.Web.Entities
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Data.Entity.Spatial;

    [Table("Ride")]
    public partial class Ride
    {
        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2214:DoNotCallOverridableMethodsInConstructors")]
        public Ride()
        {
            Comments = new HashSet<Comment>();
            UserRides = new HashSet<UserRide>();
        }

        public int RideId { get; set; }

        public int? TrajetId { get; set; }

        public int CreatorId { get; set; }

        public DateTime DateDepart { get; set; }

        public DateTime DateFin { get; set; }

        [StringLength(200)]
        public string Description { get; set; }

        public int LevelId { get; set; }

        [Required]
        [StringLength(100)]
        public string Title { get; set; }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<Comment> Comments { get; set; }

        public virtual Level Level { get; set; }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<UserRide> UserRides { get; set; }

        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
        public virtual ICollection<User> Participants { get; set; }

        public virtual Trajet Trajet { get; set; }

        public virtual User User { get; set; }
    }
}
